backend/engine/plugins/cicd_tools/interfaces/detector.py

```diff
@@ -0,0 +1,17 @@
+from typing import TypedDict
+
+
+class DetectorResult(TypedDict):
+    id: str
+    name: str
+    configs: list[str]
+    in_use: bool
+
+    debug: list[str]
+    alerts: list[str]
+    errors: list[str]
+
+
+class Detector:
+    def check(self, path: str) -> DetectorResult:
+        raise NotImplementedError()
```

Comment:
(minor) This is a good opportunity to use [`ABC`](https://docs.python.org/3/library/abc.html#abc.abstractmethod) and [`@abstractmethod`](https://docs.python.org/3/library/abc.html#abc.abstractmethod). If we add additional required methods in the future, the type-checker will warn us about missing methods.

```python
from abc import ABC, abstractmethod

class Detector(ABC):
    @abstractmethod
    def check(self, path: str) -> DetectorResult: ...
```

Example:

```python
class Foo(Detector):
    ...

foo = Foo()  # Cannot instantiate abstract class "Foo": "Detector.check" is not implemented
```